src/Integration.Vsix/Suppression/DelegateInjector.cs

```diff
@@ -61,8 +75,38 @@ private void OnAssemblyLoad(object sender, AssemblyLoadEventArgs args)
 
         private void InjectSuppressionDelegate(Assembly asm)
         {
-            // TODO: if this is a Sonar analyzer assembly, try to set the suppression delegate
+            // If this is a Sonar analyzer assembly, try to set the suppression delegate
             // Note: the property might not exist for down-level versions of the analyzer
+            const string AssemblyName = "SonarAnalyzer";
+
+            if (asm.FullName.StartsWith(AssemblyName, StringComparison.OrdinalIgnoreCase))
+            {
+                SafeSetProperty(asm);
+            }
+        }
+
+        private void SafeSetProperty(Assembly asm)
+        {
+            const string FullTypeName = "SonarAnalyzer.Helpers.SonarAnalysisContext";
+            const string PropertyName = "ShouldDiagnosticBeReported";
+
+            try
+            {
+                Type baseType = asm.GetType(FullTypeName, throwOnError: false);
+
+                PropertyInfo pi = baseType?.GetProperty(PropertyName, BindingFlags.Public | BindingFlags.Static);
+
+                pi?.SetValue(null, this.suppressionFunction);
+            }
+            catch (Exception e)
+            {
+                // Suppress failures - we don't want the analyzers to fail
+                VsShellUtils.WriteToSonarLintOutputPane(serviceProvider,
+                    $@"Unable to set the analyzer suppression handler for {asm.FullName}.
```

Comment:
[ignore] Wow first time I see this usage! Didn't even wondered myself whether it is possible to combine them.

Reply:
First time I've tried it too :-)